src/js/model/primitives/Primitive.js

```diff
@@ -31,7 +32,9 @@ function _clean(spec, clean) {
     if (c) {
       delete spec[k];
     } else if (dl.isObject(p)) {
-      spec[k] = p.signal && cln ? sg.get(p.signal) : _clean(spec[k], clean);
+      spec[k] = p.signal && cln ?
+      getIn(store.getState(), 'signals.' + p.signal + '.init') :
```

Comment:
Since we do this in two places now, I'd recommend making a `getSignal` method in the store-utils module, which takes a store and a signal name